lib/config/optionsconfiguration.go

```diff
@@ -152,3 +154,20 @@ func (opts OptionsConfiguration) GlobalDiscoveryServers() []string {
 	}
 	return util.UniqueTrimmedStrings(servers)
 }
+
+func (opts OptionsConfiguration) EffectiveMaxConcurrentFolders() int {
+	// If a value is set, trust that.
+	if opts.MaxConcurrentFolders > 0 {
+		return opts.MaxConcurrentFolders
+	}
+	// Otherwise default to the number of CPU cores in the system as a rough
+	// approximation of system powerfullness.
+	if n := runtime.GOMAXPROCS(-1); n > 0 {
+		return n
+	}
+	// We should never get here to begin with, but since we're here let's
+	// use some sort of reasonable compromise between the old "no limit" and
+	// getting nothing done... (Median number of folders out there at time
+	// of writing is two, 95-percentile at 12 folders.)
+	return 4 // https://xkcd.com/221/
```

Comment:
Perhaps worth adding a notice about this new behaviour? Or you reckon change log entry is enough?

Essentially, if I had a setup where I have 10 folders, each on a different (or simply much more capable) io subsystem, this might lead to a "wtf why is it slow" moment.

Reply:
I think change log is enough.

(And this specific case (`return 4`) might as well have been a panic, I don't think gomaxprocs ever returns <= 0, although I guess you mean the larger behavior change of defaulting to gomaxprocs.)